value of type []interp.valueInterface is not assignable to type []interface {}

yaegi version: github.com/containous/yaegi@v0.8.13-0.20200712122003-0c8f538cd973

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

// Filter is a filter
type Filter interface {
	Foo()
}

// GIFT is a gift
type GIFT struct {
	Filters []Filter
}

// New is a new filter list
func New(filters ...Filter) *GIFT {
	return &GIFT{
		Filters: filters,
	}
}

// List lists filters
func (g *GIFT) List() {
	fmt.Printf("Hello from List!\n")
}

// MyFilter is one of the filters
type MyFilter struct{}

// Foo is a foo
func (f *MyFilter) Foo() {}

func main() {
	g := New(&MyFilter{})
	g.List()
}
```

Expected result:
```console
$ go run sample.go 
Hello from List!
```

Got:
```console
$ yaegi ./sample.go 
./sample.go:18:3: panic
./sample.go:34:12: panic
reflect.Set: value of type []interp.valueInterface is not assignable to type []interface {}
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc00036ae00, 0x40)
        /usr/lib/go-1.14/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0001a1de0)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:333 +0xbc
panic(0xd2bc00, 0xc000214cb0)
        /usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003ea0a0, 0xc0003f9600)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:116 +0x20c
panic(0xd2bc00, 0xc000214cb0)
        /usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003ea140, 0xc0003f6900)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:116 +0x20c
panic(0xd2bc00, 0xc000214cb0)
        /usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
reflect.Value.assignTo(0xd1f440, 0xc0002132a0, 0x197, 0xe6805f, 0xb, 0xd1ffc0, 0x0, 0x413ce4, 0xc0001a1980, 0x0)
        /usr/lib/go-1.14/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xd1ffc0, 0xc0002131a0, 0x197, 0xd1f440, 0xc0002132a0, 0x197)
        /usr/lib/go-1.14/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp.doCompositeSparse.func1(0xc0003ea140, 0xc0002223c0)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:2112 +0x160
github.com/containous/yaegi/interp.runCfg(0xc0003f6900, 0xc0003ea140)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:122 +0x6d
github.com/containous/yaegi/interp.call.func7(0xc0003ea0a0, 0xc0003e9c70)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:1008 +0xc42
github.com/containous/yaegi/interp.runCfg(0xc0003f9600, 0xc0003ea0a0)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:122 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0002341a0, 0xc0003f8b00, 0xc0003ea000)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:100 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0002341a0, 0xc00036a600, 0x1e2, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/me/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:409 +0x52b
main.main()
        /home/me/src/softwares/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:159 +0x65d
```
May be related to https://github.com/containous/yaegi/issues/681

--- a/_test/issue-776.go
+++ b/_test/issue-776.go
@@ -1,0 +1,39 @@
+package main
+
+import "fmt"
+
+// Filter is a filter
+type Filter interface {
+	Foo()
+}
+
+// GIFT is a gift
+type GIFT struct {
+	Filters []Filter
+}
+
+// New is a new filter list
+func New(filters ...Filter) *GIFT {
+	return &GIFT{
+		Filters: filters,
+	}
+}
+
+// List lists filters
+func (g *GIFT) List() {
+	fmt.Printf("Hello from List!\n")
+}
+
+// MyFilter is one of the filters
+type MyFilter struct{}
+
+// Foo is a foo
+func (f *MyFilter) Foo() {}
+
+func main() {
+	g := New(&MyFilter{})
+	g.List()
+}
+
+// Output:
+// Hello from List!

--- a/interp/run.go
+++ b/interp/run.go
@@ -2116,6 +2116,10 @@ func doCompositeSparse(n *node, hasType bool) {
 		switch {
 		case c1.typ.cat == funcT:
 			values[field] = genFunctionWrapper(c1)
+		case c1.typ.cat == interfaceT:
+			values[field] = genValueInterfaceValue(c1)
+		case isArray(c1.typ) && c1.typ.val != nil && c1.typ.val.cat == interfaceT:
+			values[field] = genValueInterfaceArray(c1)
 		case isRecursiveType(n.typ.field[field].typ, n.typ.field[field].typ.rtype):
 			values[field] = genValueRecursiveInterface(c1, n.typ.field[field].typ.rtype)
 		default:

--- a/interp/type.go
+++ b/interp/type.go
@@ -1541,6 +1541,10 @@ func isSendChan(t *itype) bool {
 	return rt.Kind() == reflect.Chan && rt.ChanDir() == reflect.SendDir
 }
 func isMap(t *itype) bool { return t.TypeOf().Kind() == reflect.Map }
+func isArray(t *itype) bool {
+	k := t.TypeOf().Kind()
+	return k == reflect.Array || k == reflect.Slice
+}
 
 func isInterfaceSrc(t *itype) bool {
 	return t.cat == interfaceT || (t.cat == aliasT && isInterfaceSrc(t.val))

